vertical-pod-autoscaler/enhancements/8459-memory-per-cpu/README.md

```diff
@@ -0,0 +1,167 @@
+  <!-- /toc -->
+
+## Summary
+
+This AEP proposes a new feature to allow enforcing a fixed memory-per-CPU ratio (`memoryPerCPU`) in Vertical Pod Autoscaler (VPA) recommendations.  
+The feature is controlled by a new alpha feature gate `MemoryPerCPURatio` (default off).
+
+## Motivation
+
+Many workloads scale their memory requirements proportionally to CPU, but today VPA generates independent CPU and memory recommendations. This can lead to skewed configurations — for example, too much memory for a small CPU allocation, or too little memory for a high CPU allocation.
+
+The `memoryPerCPU` field addresses this by enforcing a predictable CPU-to-memory ratio in recommendations. This reduces the risk of misconfiguration, ensures consistency, and simplifies tuning for workloads where CPU and memory usage are tightly coupled.
+
+This feature is particularly useful in environments where services are billed primarily on CPU with a fixed CPU-to-memory ratio. In such cases, it allows VPA to be used for automatic vertical scaling while preserving the existing billing model and guarantees to customers.
+
+### Goals
+
+* Allow users to specify a `memoryPerCPU` ratio in `VerticalPodAutoscaler` objects.  
+* Ensure VPA recommendations respect the ratio across Target, LowerBound, UpperBound, and UncappedTarget.  
+* Provide a feature gate to enable/disable the feature cluster-wide.
+
+### Non-Goals
+
+* Redesign of the VPA recommender algorithm beyond enforcing the ratio.  
+* Supporting multiple ratio policies per container (only one `memoryPerCPU` is supported).  
+* Retroactive migration of existing VPAs without explicit user opt-in.
+
+## Proposal
+
+Extend `ContainerResourcePolicy` with a new optional field:
+
+```yaml
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: my-app
+spec:
+  resourcePolicy:
+    containerPolicies:
+      - containerName: app
+        minAllowed:
+          cpu: 1
+          memory: 4Gi
+        maxAllowed:
+          cpu: 4
+          memory: 16Gi
+        controlledResources: ["cpu", "memory"]
+        controlledValues: RequestsAndLimits
+        memoryPerCPU: "4Gi"
+```
+
+When enabled, VPA will adjust CPU or memory recommendations to maintain:
+
+```
+memory_bytes = cpu_cores * memoryPerCPU
+```
+
+## Design Details
+
+### API Changes
+
+* New field `memoryPerCPU` (`resource.Quantity`) in `ContainerResourcePolicy`.  
+* Feature gate: `MemoryPerCPURatio` (alpha, default off).
+
+### Behavior
+
+* If both CPU and memory are controlled, VPA enforces the ratio.  
+* Applies to Target, LowerBound, UpperBound, and UncappedTarget.
+* Ratio enforcement is strict:
+  * If the memory recommendation would exceed `cpu * memoryPerCPU`, then **CPU is increased** to satisfy the ratio.
+  * If the CPU recommendation would exceed `memory / memoryPerCPU`, then **memory is increased** to satisfy the ratio.
+* If ratio cannot be applied (e.g., missing CPU), fallback to standard recommendations.  
```

Comment:
I think this still needs updating, as per https://github.com/kubernetes/autoscaler/pull/8459/files#r2319103153